packages/components/atoms/f-toggle/src/components/Toggle.vue

```diff
@@ -0,0 +1,223 @@
+
+export default {
+    name: 'FToggle',
+
+    components: {
+        CheckIcon
+    },
+
+    inheritAttrs: false,
+
+    props: {
+        isModeRightToLeft: {
+            type: Boolean,
+            default: false
+        },
+
+        ariaLabelledBy: {
+            type: String,
+            default: ''
+        },
+
+        ariaDescribedBy: {
+            type: String,
+            default: ''
+        }
+    },
+
+    computed: {
+        uniqueId () {
+            const idArray = ['toggle'];
+
+            if (this.$attrs.name) idArray.push(this.$attrs.name);
+            if (this.$attrs.id) idArray.push(this.$attrs.id);
+
+            idArray.push(this.uniqueIdentifierId);
+            return idArray.join('-');
+        },
+
+        isChecked () {
+            return this.$attrs.checked;
+        }
+    },
+
+    created () {
+        this.uniqueIdentifierId = uuid();
+    },
+
+    methods: {
+        toggleOption (event) {
+            this.$emit('update', event.target.checked);
+        }
+    }
+};
+</script>
+
+<style lang="scss" module>
+@use '@justeat/fozzie/src/scss/fozzie' as f;
+
+$toggle-colour: f.$color-interactive-form;
+$toggle-colour--checked: f.$color-interactive-brand;
+$toggle-colour--disabled: f.$color-disabled-01;
+
+$toggle-width: 48px;
+$toggle-height: 24px;
+$toggle-control-size: 20px;
+$toggle-padding: 2px;
+$toggle-radius: f.$radius-rounded-e;
+$toggle-icon-size: 14px;
+
+$toggle-translation: translateX(calc($toggle-width - $toggle-control-size - 2 * $toggle-padding));
+
+@mixin toggle-transition ($property) {
+    transition-property: $property;
+    transition-duration: 0.15s;
+    transition-timing-function: cubic-bezier(0.4, 0, 0.9, 1);
+    transition-delay: 0s;
+}
+
+.c-toggle {
+    display: flex;
+
+    @include toggle-transition(background-color);
+
+    width: $toggle-width;
+    height: $toggle-height;
+    padding: $toggle-padding;
+    border-radius: $toggle-radius;
+    background-color: $toggle-colour;
+
+    cursor: pointer;
+
+    &:hover {
+        background-color: darken($toggle-colour, f.$color-hover-01);
+    }
+
+    &:focus,
+    &:focus-within {
+        background-color: $toggle-colour;
+        box-shadow: 0 0 0 2px f.$color-focus-inner, 0 0 0 4px f.$color-focus-outer;
+    }
+
+    &:active {
+        background-color: darken($toggle-colour, f.$color-active-01);
+    }
+}
+
+.c-toggle-input {
+    appearance: none;
+    margin: 0;
+}
+
+.c-toggle-control {
+    @include toggle-transition(transform);
+
+    display: flex;
+
+    align-items: center;
+    justify-content: center;
+
+    width: $toggle-control-size;
+    height: $toggle-control-size;
+
+    border-radius: $toggle-radius;
+    background-color: f.$color-content-interactive-light;
+}
+
+.c-toggle-icon {
+    width: $toggle-icon-size;
+    height: $toggle-icon-size;
+
+    path {
+        @include toggle-transition(fill);
+        fill: $toggle-colour--checked;
+    }
+}
+
+
+.c-toggle--checked {
```

Comment:
Do we need a class here: Can we use :checked?

Reply:
Not to pre-empt @FayeCarter 's response, but I'm guessing you need the class to style the parent label, as there's no way to target the parent of `:checked`?

Reply:
that's a very fair point!

Reply:
I think I've found a way. Not sure if it makes the CSS more complicated though